mobile/lib/modules/backup/background_service/background.service.dart

```diff
@@ -410,7 +414,14 @@ class BackgroundService {
     final bool ok = await backupService.backupAsset(
       toUpload,
       _cancellationToken!,
-      notifyTotalProgress ? _onAssetUploaded : (assetId, deviceId) {},
+      (assetId, deviceId, isDuplicated) {
+        if (isDuplicated) {
+          backupService.saveDuplicatedAssetIdToLocalStorage(assetId);
```

Comment:
calling the quite expensive `saveDuplicatedAssetIdToLocalStorage` method after every single assets could slow down the backup process and certainly waste precious battery resources.
We could just do it after the entire backup is done/aborted by temporarily collecting the duplicated assets in a list/set.